driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -137,6 +138,18 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             : RetryDecision.rethrow();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onRequestError(Statement statement, ConsistencyLevel cl, DriverException e, int nbRetry) {
```

Comment:
> The implementation here simply mimics the legacy behavior, which was to retry on next host.

But there is a contradiction or at least its somewhat misleading in that the RetryPolicy#onRequestError doc states *a retry should only be attempted if the request is known to be idempotent* and DefaultRetryPolicy always blindly retrying regardless of idempotence.  A user could think the guidelines in the interface javadoc could reflect the default behavior but in this case that is not true, just wondering if that's worth clarifying anywhere.